src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/PublicLicenceService.kt

```diff
@@ -0,0 +1,49 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.service
+
+import jakarta.transaction.Transactional
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.LicenceRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.resource.publicApi.model.licence.LicenceSummary
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceType
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.resource.publicApi.model.licence.LicenceStatus as PublicLicenceStatus
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.resource.publicApi.model.licence.LicenceType as PublicLicenceType
+
+@Service
+class PublicLicenceService(
+  private val licenceRepository: LicenceRepository,
+) {
+
+  @Transactional
+  fun getAllLicencesByCrn(crn: String): List<LicenceSummary> {
+    val licences = licenceRepository.findAllByCrnAndStatusCodeIn(crn, LicenceStatus.IN_FLIGHT_LICENCES)
+
+    val modelLicenceSummary = licences.map {
+      transformToPublicLicenceSummary(it, it.typeCode.mapToPublicLicenceType(), it.statusCode.mapToPublicLicenceStatus())
+    }
+
+    return modelLicenceSummary
+  }
+
+  private fun LicenceType.mapToPublicLicenceType() =
```

Comment:
the public package could have its own ToModel functions file and contain these transformer functions

Reply:
Done